src/models/neural_networks/utils.py

```diff
@@ -0,0 +1,25 @@
+import torch
+from torch import nn
+
+
+class VariationalDropout(nn.Module):
+    """
+    This ensures the same dropout is applied to each timestep,
+    as described in https://arxiv.org/pdf/1512.05287.pdf
+    """
+    def __init__(self, p):
+        super().__init__()
+
+        self.p = p
+        self.mask = None
+
+    def update_mask(self, x_shape, is_cuda):
+        mask = torch.bernoulli(torch.ones(x_shape) * (1 - self.p)) / (1 - self.p)
```

Comment:
So it's randomly choosing parts of the input matrix (`x` here does not refer to the input data - `x` static/dynamic) in the layers and applying the SAME mask to all of those layers? 

e.g. on the RNN it might apply the same mask to all of the `input_gate` matrices?